### ISSUE TITLE
NSSecureCoding instead of NSCoding

### ISSUE THREAD

**Issue by baxissimo** [google/eDistantObject#22]
Would it make sense for this project to use NSSecureCoding instead of NSCoding?
I'm not sure what all the issues are surrounding this, I just know our code base has a test that tries to make sure all the classes we depend on DO NOT use NSCoding, and it is finding the NSCoding usages in this project.

**Comment by wuhao5** [comment]
Yes. There wasn't an issue to not use `NSSecureCoding` but only for a quick serialization solution. If you have better solutions, any PR is welcomed!
